app/scripts/models/brand.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var Brand = (function () {
+
+  var Brand = function Brand(data) {
+    this.categoryIdsCsl = data.hasOwnProperty('categories') ? data.categories : null;
+    this.description = data.hasOwnProperty('description') ? data.description : null;
+    this.id = data.hasOwnProperty('id') ? parseInt(data.id) : null;
+    this.image = data.hasOwnProperty('image') ? data.image : null;
+    this.name = data.hasOwnProperty('name') ? data.name : null;
+    this.ranking = data.hasOwnProperty('ranking') ? data.ranking : null;
+    this.brandUrl = null; // @TODO: There does not seem to be a URL property in the data.
+
+    // Convert the category ids CSL to an array and remember it.
+    this.categoryIds = this.categoryIdsCsl === null ? [] : String(this.categoryIdsCsl).split(',');
+  };
+
+  Brand.prototype = {
+    getId: function getId() {
```

Comment:
nit: we should keep the order of get, has and set the same for all of these setters

Reply:
It took me a while to work out what you meant by "nit"!

But anyway, I was trying to keep them consistent as I went along, but my changes were so big it was inevitable that some adjustments would be necessary.

I've gone through all the models and made sure they follow the order "get", "has", "set".

I would have liked to have had an "abstract model" with the common properties and getters & setters in, which the models would extend, but I haven't worked out how to do it yet!